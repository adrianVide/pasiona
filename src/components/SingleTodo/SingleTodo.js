import React from "react";
import axios from "axios";
import { useState } from "react";

const SingleTodo = (props) => {
  const todo = props.todo;

  const [title, setTitle] = useState("");
  const [body, setBody] = useState("");
  
  async function handleFormSubmit(event) {
    event.preventDefault();
    try {
      await axios.put(`http://localhost:3000/todos/${todo.id}`, {
        title,
        body,
      });
      setTitle("");
      setBody("");
      props.callBack();
    } catch (error) {
      console.log(error);
    }
  }

  return (
    <li data-testid='single-todo' className="max-w-md mx-auto bg-white rounded-xl shadow-md overflow-hidden md:max-w-2xl m-4">
      <figure className="md:flex bg-gray-100 rounded-xl p-8 md:p-0">
        <img
          className="w-32 h-32 md:w-48 md:h-auto md:rounded-none rounded-full mx-auto"
          src={`https://loremflickr.com/384/512/${todo.title}`}
          alt=""
          width="384"
          height="512"
        />
        <div className="pt-6 md:p-5 text-center md:text-left space-y-4 w-full">
          <figcaption className="font-bold">
            <div className="text-gray-800">{todo.title}</div>
          </figcaption>
          <blockquote>
            <p className="text-lg font-medium text-gray-600">{todo.body}</p>
          </blockquote>

          <form onSubmit={handleFormSubmit}>
            <div>
              <input
                className="shadow appearance-none border rounded py-2 px-3 text-grey-darker m-1"
                type="text"
                placeholder="Title"
                name="title"
                value={title}
                onChange={(e) => setTitle(e.target.value)}
              />
            </div>

            <input
              className="shadow appearance-none border rounded py-2 px-3 text-grey-darker m-1"
              type="text"
              placeholder="Body"
              name="body"
              value={body}
              onChange={(e) => setBody(e.target.value)}
            />

            <input
              className="shadow appearance-none border rounded py-2 text-grey-darker mx-4 px-6"
              type="submit"
              value="Update"
            />
          </form>
        </div>
      </figure>
    </li>
  );
};

export default SingleTodo;
