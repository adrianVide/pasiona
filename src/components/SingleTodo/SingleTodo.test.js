import React from "react";
import ReactDOM from "react-dom";
import SingleTodo from "./SingleTodo";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from 'react-test-renderer'

const todo = {
  title: "Movie",
  body:
    "Movies move us like nothing else can, whether they're scary, funny, dramatic, romantic or anywhere in-between. So many titles, so much to experience.",
  id: 26,
};

afterEach(cleanup)

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<li></li>, div);
});

it("renders SingleTodo with props correctly and checks title by ID", () => {
  const { getByTestId } = render(<SingleTodo todo={todo}></SingleTodo>);
  expect(getByTestId("single-todo")).toHaveTextContent("Movie");
});

it("renders SingleTodo with props correctly and checks body by text", () => {
  const { getByText } = render(<SingleTodo todo={todo}></SingleTodo>);
  expect(
    getByText(
      "Movies move us like nothing else can, whether they're scary, funny, dramatic, romantic or anywhere in-between. So many titles, so much to experience."
    )
  ).toBeVisible;
});

it("matches snapshot", () => {
  const compTree = renderer.create(<SingleTodo todo={todo}></SingleTodo>).toJSON();
  expect(compTree).toMatchSnapshot();
})