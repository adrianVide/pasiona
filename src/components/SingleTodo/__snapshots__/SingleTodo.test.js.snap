// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<li
  className="max-w-md mx-auto bg-white rounded-xl shadow-md overflow-hidden md:max-w-2xl m-4"
  data-testid="single-todo"
>
  <figure
    className="md:flex bg-gray-100 rounded-xl p-8 md:p-0"
  >
    <img
      alt=""
      className="w-32 h-32 md:w-48 md:h-auto md:rounded-none rounded-full mx-auto"
      height="512"
      src="https://loremflickr.com/384/512/Movie"
      width="384"
    />
    <div
      className="pt-6 md:p-5 text-center md:text-left space-y-4 w-full"
    >
      <figcaption
        className="font-bold"
      >
        <div
          className="text-gray-800"
        >
          Movie
        </div>
      </figcaption>
      <blockquote>
        <p
          className="text-lg font-medium text-gray-600"
        >
          Movies move us like nothing else can, whether they're scary, funny, dramatic, romantic or anywhere in-between. So many titles, so much to experience.
        </p>
      </blockquote>
      <form
        onSubmit={[Function]}
      >
        <div>
          <input
            className="shadow appearance-none border rounded py-2 px-3 text-grey-darker m-1"
            name="title"
            onChange={[Function]}
            placeholder="Title"
            type="text"
            value=""
          />
        </div>
        <input
          className="shadow appearance-none border rounded py-2 px-3 text-grey-darker m-1"
          name="body"
          onChange={[Function]}
          placeholder="Body"
          type="text"
          value=""
        />
        <input
          className="shadow appearance-none border rounded py-2 text-grey-darker mx-4 px-6"
          type="submit"
          value="Update"
        />
      </form>
    </div>
  </figure>
</li>
`;
